src/main/java/com/rackspace/salus/telemetry/etcd/workpart/WorkAllocator.java

```diff
@@ -664,48 +631,69 @@ private void processStoppedWork(String workId, String releasedContent) {
         .get(
             fromString(prefix + REGISTRY_SET + workId)
         )
-        .thenApply(getResponse -> getResponse.getKvs().get(0).getValue().toString(StandardCharsets.UTF_8));
+        .thenApply(getResponse -> getResponse.getKvs().isEmpty() ?
+            null : valueAsString(getResponse.getKvs().get(0)));
+  }
+
+  private CompletableFuture<Long> getCurrentWorkLoad() {
+    final ByteSequence activeSetKey = fromString(prefix + ACTIVE_SET);
+
+    return etcd.getKVClient()
+        .get(
+            activeSetKey,
+            GetOption.newBuilder()
+                .withPrefix(activeSetKey)
+                .build()
+            )
+        .thenApply(getResponse ->
+            getResponse.getKvs().stream()
+            .filter(keyValue -> valueAsString(keyValue).equals(ourId))
+            .count());
   }
 
   private CompletableFuture<Boolean> amILeastLoaded(long atRevision) {
-    // Because of the zero-padded formatting of the work load value stored in each worker entry,
-    // we can find the least loaded worker via etcd by doing an ASCII sort of the values and picking
-    // off the lowest value.
+    final ByteSequence activeSetKey = fromString(prefix + ACTIVE_SET);
+
+    // Iterate over a snapshot of the active entries, where the value of each is a worker ID...
     return etcd.getKVClient()
         .get(
-            fromString(prefix + WORKERS_SET),
+            activeSetKey,
             GetOption.newBuilder()
-                .withPrefix(fromString(prefix + WORKERS_SET))
-                .withSortField(SortTarget.VALUE)
-                .withSortOrder(SortOrder.ASCEND)
-                .withLimit(1)
+                .withPrefix(activeSetKey)
                 .withRevision(atRevision)
                 .build()
         )
         .thenApply(getResponse -> {
-          if (getResponse.getCount() <= 1) {
-            log.debug("Skipping least-loaded evaluation since I'm the only worker");
-            // it's only us, so we're it
+          // Group by worker ID and count each
+          final Map<String, Long> grouped = getResponse.getKvs().stream()
+              // ...extract just the worker ID
+              .map(Bits::valueAsString)
+              // ...group and count
+              .collect(Collectors.groupingBy(s -> s, Collectors.counting()))
```

Comment:
```suggestion
              .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()))
```
I think using this is slightly more efficient.

Reply:
Thanks, I knew such a thing existed, but couldn't find it. I was hoping IntelliJ would suggest the fix, but it let me down.

Reply:
Well, it's way down in the list

![image](https://user-images.githubusercontent.com/988985/89072501-52466680-d33e-11ea-8363-98d0d148d60d.png)